### DISCUSSION TITLE
Considering migrating to FreeBSD/pooudriere from Linux/Buildroot

### DISCUSSION THREAD

**Discussion by ecorm** [freebsd/poudriere#1056] [Q&A]
I'm working on a project using the BeagleBone Black, and I've been able to successfully cross-compile *complete* Linux images via [Buildroot](https://buildroot.org/).

These generated images include:
- the bootloader
- the Linux kernel
- userland library/application packages (sshd, nginx, etc)
- our compiled application executable (which is built within Buildroot)
- custom configuration files for things such as nginx

All this can be generated from a single invocation of Buildroot. There is no need to manually tinker with an intermediate image to obtain the desired final result. Furthermore, I can choose a pre-built kernel image, which saves much time when building from scratch.

Can all this be done with poudriere?

This post is not meant to bash FreeBSD or poudriere. It's just that my application is a commercial one, and the licensing terms of FreeBSD saves me some legal headaches. But if I have to give up the convenience of being able to generate a complete minimalist target image in ~10 minutes, then I might consider sticking with Linux. We are not making any modifications to GPL'ed code, and our closed-source application executable does not statically link to GPL or LGPL libraries.

**Comment by dch** [comment]
@ecorm this is very easy on FreeBSD with poudriere, https://github.com/freebsd/poudriere/wiki/poudriere-image.8-devel if you contact me va email I can share some applications using this.

**Comment by jlduran** [comment]
Also, take a look at:
https://github.com/jlduran/router
https://github.com/jlduran/rpi (arm64-based)
As a starting point for inspiration.
